### PR TITLE
Updated Swagger Schema and Tests for MS Peering

### PR DESCRIPTION
**Pull request by rileymckenna**
Updated Data model for Peering Object and test to accompany.

### CODE REVIEW

**Comment by rileymckenna** [comment]
updated and made the necessary changes

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -832,7 +832,6 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/ManagementConfigurations" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/solutions" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings_prefixes" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >peerings_prefixes [](start = 128, length = 17)

this shouldn't be removed. I can see the resource type in Microsoft.Peering.json.

**Comment by rileymckenna** [comment]
@TianoMS Updated the schema to match the requests.

File: schemas/2019-03-01/Microsoft.Peering.json [hunk 2]
```diff
@@ -0,0 +1,405 @@
+{
+  "id": "https://schema.management.azure.com/schemas/2019-03-01/Microsoft.Peering.json#",
```
Comment (TianoMS) [inline schemas/2019-03-01/Microsoft.Peering.json]: >2019-03-01 [](start = 53, length = 10)

shouldn't this be 2019-03-01-preview?

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 3]
```diff
@@ -853,8 +853,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationalInsights.json#/resourceDefinitions/workspaces_linkedServices" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/ManagementConfigurations" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/solutions" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings_prefixes" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2019-03-01/Microsoft.Peering.json#/resourceDefinitions/peerings" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >2019-03-01 [](start = 73, length = 10)

this should be 2019-03-01-preview, right?

File: schemas/2015-01-01/deploymentTemplate.json [hunk 4]
```diff
@@ -881,8 +881,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationalInsights.json#/resourceDefinitions/workspaces_linkedServices" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/ManagementConfigurations" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/solutions" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings_prefixes" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2019-03-01-preview/Microsoft.Peering.json#/resourceDefinitions/peering" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >peering [](start = 136, length = 7)

peerings

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 5]
```diff
@@ -869,8 +869,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationalInsights.json#/resourceDefinitions/workspaces_linkedServices" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/ManagementConfigurations" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/solutions" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings_prefixes" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2019-03-01-preview/Microsoft.Peering.json#/resourceDefinitions/peering" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >peering [](start = 136, length = 7)

peerings

File: schemas/2015-01-01/deploymentTemplate.json (hunk 4 above)
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >{ "$ref": "https://schema.management.azure.com/schemas/2019-03-01-preview/Microsoft.Peering.json#/resourceDefinitions/peering" }, [](start = 18, length = 129)

can you also update in 2019-04-01/deploymentTemplate.json?

**Comment by TianoMS** [comment]
>                   { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings_prefixes" },

guess you need to remove them. 

---
Refers to: schemas/2019-04-01/deploymentTemplate.json:883 in 5c6e728. [](commit_id = 5c6e728f5f00624a2d0e629b0d61cec41325a27f, deletion_comment = False)

File: schemas/2019-04-01/deploymentTemplate.json [hunk 6]
```diff
@@ -879,7 +880,6 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/ManagementConfigurations" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01-preview/Microsoft.OperationsManagement.json#/resourceDefinitions/solutions" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings" },
```
Comment (TianoMS) [inline schemas/2019-04-01/deploymentTemplate.json]: >{ "$ref": "https://schema.management.azure.com/schemas/2018-10-01/Microsoft.Peering.json#/resourceDefinitions/peerings" }, [](start = 18, length = 122)

this one needs to be removed too.

**Comment by Tiano2017** [comment]
cheers. the validations passed. Thanks for making the changes.

**Pull request merged**